app/Http/Controllers/CourseMaterialController.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+namespace Interns2024c\Http\Controllers;
+
+use Illuminate\Http\Request;
+
+class CourseMaterialController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     */
+    public function index()
+    {
+        $materials = \Interns2024c\Models\CourseMaterial::with('lesson')->get();
+
+        $englishLevels = ['English A1', 'English A2', 'English B1', 'English B2', 'English C1', 'English C2', 'No Lesson'];
+
+        $groupedMaterials = $materials->groupBy(function ($material) {
+            return $material->lesson->title ?? 'No Lesson';
+        });
+
+        $sortedMaterials = $groupedMaterials->sortBy(function ($group, $lessonTitle) use ($englishLevels) {
+            $index = array_search($lessonTitle, $englishLevels);
+            return $index !== false ? $index : PHP_INT_MAX;
+        });
+
+        $transformedMaterials = $sortedMaterials->map(function ($group, $lessonTitle) {
+            return [
+                'lesson_title' => $lessonTitle,
+                'materials' => $group->map(function ($material) {
+                    return [
+                        'id' => $material->id,
+                        'title' => $material->title,
+                        'description' => $material->description,
+                        'file_path' => $material->file_path,
+                        'type' => $material->type,
+                    ];
+                })->values(),
+            ];
+        })->values();
+
+        return inertia('CourseMaterials/Index', [
+            'materials' => $transformedMaterials,
+        ]);
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     */
+    public function create()
+    {
+        $lessons = \Interns2024c\Models\Lesson::all();
+
+        return inertia('CourseMaterials/Create', [
+            'lessons' => $lessons,
+        ]);
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     */
+    public function store(Request $request)
+    {
+        $validated = $request->validate([
+            'title' => 'required|string|max:255',
+            'type' => 'required|string|in:text,file,video',
+            'description' => 'nullable|string',
+            'lesson_id' => 'required|exists:lessons,id',
+            'file' => 'nullable|file|mimes:jpg,png,pdf,mp4|max:20480',
+        ]);
+
+        if ($request->hasFile('file')) {
+            $filePath = $request->file('file')->store('uploads', 'public');
+            $validated['file_path'] = $filePath;
+        }
+
+        \Interns2024c\Models\CourseMaterial::create($validated);
+
+        return redirect()->route('course-materials.index')->with('success', 'Course material created successfully.');
+    }
+
+    /**
+     * Display the specified resource.
+     */
+    public function show($id)
```

Comment:
Missing type and return type - this is not the only place in the code where this occurs. Please review the rest of the code and make the necessary corrections everywhere.